satellite/audit/reporter.go

```diff
@@ -36,6 +36,10 @@ type Report struct {
 
 // NewReporter instantiates a reporter
 func NewReporter(log *zap.Logger, overlay *overlay.Cache, containment Containment, maxRetries int, maxReverifyCount int32) *Reporter {
+	if log == nil {
+		log = log.Named("audit:reporter")
```

Comment:
This shouldn't be done, the logger must come from outside, if it does not then it is a bug.

This check also does not make sense.

Reply:
I think it should at minimum be a noop ... I get it's a bug if a logger isn't passed but we shouldn't allow the system to panic

Reply:
No, this is completely a programmer bug that should never pass tests.

Reply:
it was a comment/suggestion I received from peer review.  I will reopen the branch to update it.